azurerm/internal/services/cognitive/cognitive_account_resource.go

```diff
@@ -146,7 +227,46 @@ func resourceCognitiveAccount() *pluginsdk.Resource {
 				},
 			},
 
-			"custom_subdomain_name": {
+			"public_network_access_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"restrict_outbound_network_access": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"storage": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"resource_id": {
```

Comment:
what type of resource is this?

Reply:
storage account id, I think it's better to rename it to `storage_account_id`. I'll change that.